### ISSUE TITLE
Cache frequently accessed structures in `PartitionStore`

### ISSUE THREAD

**Issue by tillrohrmann** [restatedev/restate#1873]

**Comment by slinkydeveloper** [comment]
Could this LRU cache be completely hidden behind the `InvocationStatusTable` impl? In particular once we get rid of these https://github.com/restatedev/restate/issues/276

**Comment by tillrohrmann** [comment]
I've ran some simple experiments and caching of the `InvocationStatus` didn't have a measurable impact on throughput.

**Comment by slinkydeveloper** [comment]
> I've ran some simple experiments and caching of the InvocationStatus didn't have a measurable impact on throughput.

Did you by any chance tried with the neo invocation status to see if the new protobuf layout has any either positive or negative impact?

**Comment by tillrohrmann** [comment]
> Did you by any chance tried with the neo invocation status to see if the new protobuf layout has any either positive or negative impact?

No, I only tried it with the old InvocationStatus. Given that reading the `InvocationStatus` took ~1.5% of time in the flamegraphs, I wouldn't expect a super noticeable change. This doesn't mean that the we shouldn't optimize these things, though.
